src/Datasource/EntityTrait.php

```diff
@@ -500,80 +512,91 @@ public function getVirtual(): array
     }
 
     /**
-     * Get the list of visible properties.
+     * Gets the list of visible fields.
      *
-     * The list of visible properties is all standard properties
-     * plus virtual properties minus hidden properties.
+     * The list of visible fields is all standard fields
+     * plus virtual fields minus hidden fields.
      *
-     * @return array A list of properties that are 'visible' in all
+     * @return array A list of fields that are 'visible' in all
      *     representations.
      */
-    public function visibleProperties()
+    public function getVisible(): array
     {
-        $properties = array_keys($this->_properties);
-        $properties = array_merge($properties, $this->_virtual);
+        $fields = array_keys($this->_fields);
+        $fields = array_merge($fields, $this->_virtual);
+
+        return array_diff($fields, $this->_hidden);
+    }
 
-        return array_diff($properties, $this->_hidden);
+    /**
+     * Gets the list of visible fields.
+     *
+     * @deprecated 4.0.0 Use getVisible() instead. Will be removed in 5.0.
+     * @return array
+     */
+    public function visibleProperties(): array
+    {
+        return $this->getVisible();
```

Comment:
Should this trigger a warning?

Reply:
We have added any warnings for new deprecations in 4.0.

Reply:
Not inside 4.0.0 IMO, I guess these kinds of things could be useful more towards 4.1, 4.2 when starting to prepare for 5.0 migration.

Reply:
I guess so :)